plugins/modules/foreman_scc_product.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Manisha Singhal (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_scc_product
+short_description: Subscribe Foreman Suse Customer Center Account Product
+description:
+  - "Manage Foreman Suse Customer Center Product Entities"
+  - "This module requires the foreman_scc_manager plugin set up in the server"
+  - "See: U(https://github.com/ATIX-AG/foreman_scc_manager)"
+author:
+  - "Manisha Singhal (@manisha15) ATIX AG"
+options:
+  friendly_name:
```

Comment:
I am a tad confused about the API here. You can search on `name` and on `friendly_name` (https://github.com/ATIX-AG/foreman_scc_manager/blob/67dc908192f729b2de90169188def6b4614e5aa0/app/models/scc_product.rb#L21-L22) but only `friendly_name` is unique?

What I am wondering is: would it hurt if we call this Ansible param `name`, even tho it really represents `friendly_name`? It just feels weird to use something else but `name` (which is a very common Ansible pattern) to find something.

Reply:
Yeah, it is confusing. Even the information that `friendly_name` is unique (enough) is only evidence based. But renaming it to `name` could be even more confusing if someone had a look at the the api output. Keeping `friendly_name` might help to ring the right bells.

Reply:
k, I won't have the pleasure to use the module any time soon, so I'll trust your judgement :)